pkg/app/cockroachdb.go

```diff
@@ -0,0 +1,312 @@
+}
+
+// Initialize is used to initialize the database or create schema
+func (c *CockroachDB) Initialize(ctx context.Context) error {
+	createDatabaseCMD := "./cockroach sql --certs-dir=/cockroach/cockroach-client-certs -e 'CREATE DATABASE bank; CREATE TABLE bank.accounts (id INT, balance DECIMAL);'"
+	createDatabase := []string{"sh", "-c", createDatabaseCMD}
+	_, stderr, err := c.execCommand(ctx, createDatabase)
+	if err != nil {
+		return errors.Wrapf(err, "Error while initializing: %s", stderr)
+	}
+	return nil
+}
+
+func (c *CockroachDB) Insert(ctx context.Context) error {
+	log.Print("Inserting some records in  cockroachdb instance.", field.M{"app": c.name})
+
+	insertRecordCMD := "./cockroach sql --certs-dir=/cockroach/cockroach-client-certs -e 'INSERT INTO bank.accounts VALUES (1, 1000.50);'"
+	insertRecord := []string{"sh", "-c", insertRecordCMD}
+	_, stderr, err := c.execCommand(ctx, insertRecord)
+	if err != nil {
+		return errors.Wrapf(err, "Error while inserting the data into database: %s", stderr)
+	}
+
+	log.Print("Successfully inserted records in the application.", field.M{"app": c.name})
+	return nil
+}
+
+func (c *CockroachDB) Count(ctx context.Context) (int, error) {
+	log.Print("Counting the records from the cockroachdb instance.", field.M{"app": c.name})
+
+	selectRowsCMD := "./cockroach sql --certs-dir=/cockroach/cockroach-client-certs -e 'SELECT COUNT(*) FROM bank.accounts;'"
+	selectRows := []string{"sh", "-c", selectRowsCMD}
+	stdout, stderr, err := c.execCommand(ctx, selectRows)
+	if err != nil {
+		return 0, errors.Wrapf(err, "Error while counting the data of the database: %s", stderr)
+	}
+	// get the returned count and convert it to int, to return
+	rowsReturned, err := strconv.Atoi(strings.Split(stdout, "\n")[1])
+	if err != nil {
+		return 0, errors.Wrapf(err, "Error while converting row count to int.")
+	}
+	log.Print("Count that we received from application is.", field.M{"app": c.name, "count": rowsReturned})
+	return rowsReturned, nil
+}
+
+func (c *CockroachDB) Reset(ctx context.Context) error {
+	timeoutCtx, waitCancel := context.WithTimeout(ctx, cReadyTimeout)
+	defer waitCancel()
+	err := poll.Wait(timeoutCtx, func(ctx context.Context) (bool, error) {
+		err := c.Ping(ctx)
+		return err == nil, nil
+	})
+
+	if err != nil {
+		return errors.Wrapf(err, "Error waiting for application %s to be ready to reset it", c.name)
+	}
+
+	log.Print("Resetting the cockroachdb instance.", field.M{"app": "cockroachdb"})
+
+	// delete all the data from the table
+	deleteFromTableCMD := "./cockroach sql --certs-dir=/cockroach/cockroach-client-certs -e 'DROP DATABASE IF EXISTS bank;'"
+	deleteFromTable := []string{"sh", "-c", deleteFromTableCMD}
+	_, stderr, err := c.execCommand(ctx, deleteFromTable)
+	if err != nil {
+		return errors.Wrapf(err, "Error while dropping the table: %s", stderr)
+	}
+
+	err = poll.Wait(timeoutCtx, func(ctx context.Context) (bool, error) {
+		err = c.waitForGC(ctx)
+		return err == nil, nil
+	})
```

Comment:
```suggestion
	err = poll.Wait(timeoutCtx, func(ctx context.Context) (bool, error) {
	        // Even though the table is deleted from the database, it's present in the
	        // descriptor tables. We will have to wait for it to be deleted from there  as
	        // well (using garbage collcetion), so that we can restore the snapshot in
	        // the same DB cluster.
		err = c.waitForGC(ctx)
		return err == nil, nil
	})
```